File: LessonLink/LessonLink.Domain/DTOs/User/TeacherUpdateDto.cs
using Microsoft.AspNetCore.Http;

namespace LessonLink.BusinessLogic.DTOs.User;

public class TeacherUpdateDto
{
    public string? NickName { get; set; }
    public IFormFile? ProfilePicture { get; set; }
    public bool? AcceptsOnline { get; set; }
    public bool? AcceptsInPerson { get; set; }
    public string? Location { get; set; }
    public int? HourlyRate { get; set; }
    public string? Description { get; set; }
    public string? Contact { get; set; }
    public List<string>? SubjectNames { get; set; }
}

File: lessonlink-react-app/src/dtos/TeacherUpdateDto.ts
export type TeacherUpdateDto = {
    nickName?: string;
    profilePicture?: File;
    acceptsOnline?: boolean;
    acceptsInPerson?: boolean;
    location?: string;
    hourlyRate?: number;
    description?: string;
    contact?: string;
    subjectNames?: string[];
};
